### PR TITLE
Update bundle-fairy to accept bundles with KML/GPX files #8

### PR DESCRIPTION
**Pull request by who8mycakes**
Implementation of https://github.com/mapbox/bundle-fairy/issues/8.

This PR updates bundle-fairy to recognize and accept bundles which include archived KML and GPX files, following a pending update to preprocessorcerer: https://github.com/mapbox/preprocessorcerer/pull/75.

This PR:
* Adds .kml and .gpx file extensions to the array of allowed files.
* Adds zipped fixtures containing a kml/gpx against which the valid-bundle test is run.

cc @GretaCB

### CODE REVIEW

**Comment by GretaCB** [comment]
@who8mycakes Implementation looks 👍 

One thing to think about is how this change should affect [the specification](https://github.com/mapbox/bundle-fairy#a-brief-specification) of a bundle for KML/GPX files: 
- Should archived original KML/GPX files be mandatory or optional?
- What are the considerations for both cases?

Some things to consider:
- What is the purpose of bundle-fairy?
- How might other developers use it outside of Mapbox?

Once these decisions are ironed out:
- [ ] Make code changes that reflect the spec updates, if needed
- [ ] Update documentation

File: test/fixtures/index.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ module.exports = {
     geojson_withmetadata_noindex: path.join(__dirname, 'bundle_geojson-without-indices-with-metadata.zip'),
     single_csv_withindex: path.join(__dirname, 'bundle_single-csv-with-index.zip'),
     single_geojson_withindex: path.join(__dirname, 'bundle_single-geojson-with-index.zip'),
-    single_geojson_with_metadata: path.join(__dirname, 'bundle_single-geojson-with-metadata-without-index.zip')
+    single_geojson_with_metadata: path.join(__dirname, 'bundle_single-geojson-with-metadata-without-index.zip'),
+    with_archived_kml: path.join(__dirname, 'bundle_with-archived-kml.zip'),
+    with_archived_gpx: path.join(__dirname, 'bundle_with-archived-gpx.zip')
   },
   invalid: {
```
Comment (mapsam) [inline test/fixtures/index.js]: Are there any invalid test cases we can add? Maybe something with an incorrectly named original file?

File: index.js [hunk 2]
```diff
@@ -141,6 +142,11 @@ function isbundle(zipfile, callback) {
       return callback(null, true);
     }
 
+    //bundle: converted from gpx/kml (1 to n geojsons) and has metadata.json file and archived gpx/kml file
+    if (has_archivedfile && has_metadata && unique_extensions.indexOf('.geojson') > -1) {
```
Comment (mapsam) [inline index.js]: Will valid bundles with valid archived original ever get here? Seeing a the [check above](https://github.com/mapbox/bundle-fairy/pull/9/files#diff-168726dbe96b3ce427e7fedce31bb0bcR141) just checks for `has_metadata` and `.geojson` extensions. A bundle with an archived file will evaluate to true there, and return.

File: test/fixtures/index.js (hunk 1 above)
Comment (who8mycakes) [inline test/fixtures/index.js]: The different invalid permutations are covered by the invalid cases here: https://github.com/mapbox/bundle-fairy/blob/master/test/fixtures/index.js

The other cases cover missing metadata and indices, which would be the same invalid cases for the changes I made.

File: index.js (hunk 2 above)
Comment (who8mycakes) [inline index.js]: Thanks for pointing this out. The second check is redundant since having an archived file is optional. I've removed this check.

**Pull request merged**